File: agents/backend/onyx/server/features/quality_control_ai/utils/format_utils.py
"""
Format Utilities

Utility functions for formatting data.
"""

from typing import Any, Optional
from datetime import datetime


def format_number(
    number: float,
    decimals: int = 2,
    thousands_sep: str = ","
) -> str:
    """
    Format number with decimals and thousands separator.
    
    Args:
        number: Number to format
        decimals: Number of decimal places
        thousands_sep: Thousands separator
    
    Returns:
        Formatted string
    """
    return f"{number:,.{decimals}f}".replace(",", thousands_sep)


def format_percentage(
    value: float,
    decimals: int = 2
) -> str:
    """
    Format value as percentage.
    
    Args:
        value: Value to format (0.0 to 1.0 or 0 to 100)
        decimals: Number of decimal places
    
    Returns:
        Formatted percentage string
    """
    # If value is between 0 and 1, multiply by 100
    if 0 <= value <= 1:
        value = value * 100
    
    return f"{value:.{decimals}f}%"


def format_currency(
    amount: float,
    currency: str = "USD",
    decimals: int = 2
) -> str:
    """
    Format amount as currency.
    
    Args:
        amount: Amount to format
        currency: Currency code
        decimals: Number of decimal places
    
    Returns:
        Formatted currency string
    """
    symbols = {
        "USD": "$",
        "EUR": "€",
        "GBP": "£",
        "JPY": "¥",
    }
    symbol = symbols.get(currency, currency)
    return f"{symbol}{amount:,.{decimals}f}"


def format_datetime_human(
    dt: datetime,
    format_type: str = "short"
) -> str:
    """
    Format datetime in human-readable format.
    
    Args:
        dt: Datetime to format
        format_type: Format type ('short', 'long', 'relative')
    
    Returns:
        Formatted string
    """
    if format_type == "short":
        return dt.strftime("%Y-%m-%d %H:%M")
    elif format_type == "long":
        return dt.strftime("%Y-%m-%d %H:%M:%S")
    elif format_type == "relative":
        from .date_utils import time_ago
        return time_ago(dt)
    else:
        return dt.isoformat()


def format_list(
    items: list,
    separator: str = ", ",
    max_items: Optional[int] = None
) -> str:
    """
    Format list as string.
    
    Args:
        items: List to format
        separator: Separator string
        max_items: Maximum items to show (None for all)
    
    Returns:
        Formatted string
    """
    if max_items and len(items) > max_items:
        shown = items[:max_items]
        return separator.join(str(item) for item in shown) + f" ... ({len(items)} total)"
    return separator.join(str(item) for item in items)

File: Frontier-Model-run-polyglot/scripts/truthgpt-model-builder/components/ChatInterface/utils/formatUtils.ts
/**
 * Utility functions for formatting
 */

/**
 * Format file size
 */
export function formatFileSize(bytes: number): string {
  if (bytes === 0) return '0 Bytes'

  const k = 1024
  const sizes = ['Bytes', 'KB', 'MB', 'GB']
  const i = Math.floor(Math.log(bytes) / Math.log(k))

  return Math.round(bytes / Math.pow(k, i) * 100) / 100 + ' ' + sizes[i]
}

/**
 * Format duration
 */
export function formatDuration(seconds: number): string {
  if (seconds < 60) {
    return `${Math.floor(seconds)}s`
  }

  const minutes = Math.floor(seconds / 60)
  const remainingSeconds = Math.floor(seconds % 60)

  if (minutes < 60) {
    return `${minutes}m ${remainingSeconds}s`
  }

  const hours = Math.floor(minutes / 60)
  const remainingMinutes = minutes % 60

  return `${hours}h ${remainingMinutes}m ${remainingSeconds}s`
}

/**
 * Format relative time
 */
export function formatRelativeTime(timestamp: number): string {
  const now = Date.now()
  const diff = now - timestamp
  const seconds = Math.floor(diff / 1000)
  const minutes = Math.floor(seconds / 60)
  const hours = Math.floor(minutes / 60)
  const days = Math.floor(hours / 24)

  if (seconds < 60) {
    return 'hace un momento'
  }

  if (minutes < 60) {
    return `hace ${minutes} ${minutes === 1 ? 'minuto' : 'minutos'}`
  }

  if (hours < 24) {
    return `hace ${hours} ${hours === 1 ? 'hora' : 'horas'}`
  }

  if (days < 7) {
    return `hace ${days} ${days === 1 ? 'día' : 'días'}`
  }

  return new Date(timestamp).toLocaleDateString()
}

/**
 * Format number with commas
 */
export function formatNumber(num: number): string {
  return num.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')
}

/**
 * Format percentage
 */
export function formatPercentage(value: number, total: number, decimals: number = 1): string {
  if (total === 0) return '0%'
  const percentage = (value / total) * 100
  return `${percentage.toFixed(decimals)}%`
}

/**
 * Format currency
 */
export function formatCurrency(amount: number, currency: string = 'USD'): string {
  return new Intl.NumberFormat('es-ES', {
    style: 'currency',
    currency,
  }).format(amount)
}

/**
 * Truncate text with ellipsis
 */
export function truncateText(text: string, maxLength: number, suffix: string = '...'): string {
  if (text.length <= maxLength) return text
  return text.substring(0, maxLength - suffix.length) + suffix
}

/**
 * Format code block
 */
export function formatCodeBlock(code: string, language: string = ''): string {
  return `\`\`\`${language}\n${code}\n\`\`\``
}

/**
 * Format markdown link
 */
export function formatMarkdownLink(text: string, url: string): string {
  return `[${text}](${url})`
}

/**
 * Format list items
 */
export function formatList(items: string[], ordered: boolean = false): string {
  return items
    .map((item, index) => {
      const prefix = ordered ? `${index + 1}. ` : '- '
      return `${prefix}${item}`
    })
    .join('\n')
}

/**
 * Format table
 */
export function formatTable(headers: string[], rows: string[][]): string {
  const headerRow = `| ${headers.join(' | ')} |`
  const separatorRow = `| ${headers.map(() => '---').join(' | ')} |`
  const dataRows = rows.map(row => `| ${row.join(' | ')} |`).join('\n')

  return [headerRow, separatorRow, dataRows].join('\n')
}

/**
 * Format JSON with indentation
 */
export function formatJSON(obj: any, indent: number = 2): string {
  return JSON.stringify(obj, null, indent)
}

/**
 * Format date
 */
export function formatDate(date: Date | number, format: 'short' | 'long' | 'time' = 'short'): string {
  const d = date instanceof Date ? date : new Date(date)

  if (format === 'time') {
    return d.toLocaleTimeString('es-ES')
  }

  if (format === 'long') {
    return d.toLocaleDateString('es-ES', {
      year: 'numeric',
      month: 'long',
      day: 'numeric',
      hour: '2-digit',
      minute: '2-digit',
    })
  }

  return d.toLocaleDateString('es-ES', {
    year: 'numeric',
    month: 'short',
    day: 'numeric',
  })
}

/**
 * Format bytes to human readable
 */
export function formatBytes(bytes: number, decimals: number = 2): string {
  if (bytes === 0) return '0 Bytes'

  const k = 1024
  const dm = decimals < 0 ? 0 : decimals
  const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB']

  const i = Math.floor(Math.log(bytes) / Math.log(k))

  return parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i]
}




